File: ruby/lib/game_machine/actor_factory.rb
module GameMachine
  class ActorFactory
    include UntypedActorFactory

    def initialize(klass,args)
      @klass = klass
      @args = args
    end

    def create
      actor = @klass.new
      if @args.size >= 1 && actor.respond_to?(:post_init)
        actor.post_init(*@args)
      end
      actor
    end

    def self.create
    end
  end
end

File: server/java/server/src/main/java/io/gamemachine/core/ActorFactory.java
package io.gamemachine.core;

import akka.actor.Actor;
import akka.actor.IndirectActorProducer;
import akka.actor.UntypedActor;

class ActorFactory implements IndirectActorProducer {
    final String beanName;
    private Object factory;

    public ActorFactory(Object factory, String beanName) {
        this.beanName = beanName;
        this.factory = factory;

    }

    @Override
    public Class<? extends Actor> actorClass() {
        return UntypedActor.class;
    }

    @Override
    public UntypedActor produce() {
        return ((IActorFactory) factory).create();
    }
}